docs/pages/blog/callback-support-in-style-overrides.md

```diff
@@ -0,0 +1,189 @@
+---
+title: Introducing callback support in style overrides
+description: We're excited to introduce the callback support for global theme overrides in this minor version update!
+date: 2022-01-20T00:00:00.000Z
+authors: ['siriwatknp']
+tags: ['MUI Core', 'News']
+card: false
+---
+
+MUI Core [v5.3.0](https://github.com/mui-org/material-ui/releases/tag/v5.3.0) introduces the ability to write a callback in style overrides (the global theming), giving you full control of component customization at the theme level.
+
+Why using a callback is better than the existing plain object? Let me explain from the beginning.
+
+## The problems
+
+In v4, the style engine library was JSS which had some limitations.
+Style overrides were not able to support dynamic props via a callback so we relied on using classes. Take a look at the [`Chip` classes](https://github.com/mui-org/material-ui/blob/97d32b0ff3fae4537c20c79e619f132f4a5c5cbb/packages/mui-material/src/Chip/chipClasses.ts) for example, there are more than 20 classes that are incomplete if we count the permutation of elements (`root | avatar | icon | label | deleteIcon`), size (`small | medium | large`), and color (`primary | secondary | ...`).
+This leads to a bad theming experience because developers need to know the specific key to customize.
+
+We believe it would be better for developers if they could create custom styles by reading the component props, without ever needing to know what key they should use.
+
+Fortunately, it is now possible in v5 because of the new style engine powered by emotion. Theming is simpler and more flexible. You only need to know the component's slot name and then provide an **object** (static overrides) or a **callback** (dynamic overrides).
+
+## Using callback in `styleOverrides`
+
+The callback gives you the `props` that the slot received. Most of the time, you would use:
+
+- `props.ownerState`: the combination of runtime props and internal states.
+- `props.theme`: the object you provide to `ThemeProvider` or the default one.
+
+```jsx
+import { ThemeProvider, createTheme } from '@mui/material/styles';
+
+<ThemeProvider
+  theme={createTheme({
+    components: {
+      MuiChip: {
+        styleOverrides: {
+          // you can now use the theme without creating the initial theme!
+          root: ({ ownerState, theme }) => ({
+            padding: {
+              small: '8px 4px',
+              medium: '12px 6px',
+              large: '16px 8px',
+            }[ownerState.size],
+            ...(ownerState.variant === 'outlined' && {
+              borderWidth: '2px',
+              ...(ownerState.variant === 'primary' && {
+                borderColor: theme.palette.primary.light,
```

Comment:
OT: How circular references prevented/resolved?

Reply:
Sorry, I don't quite understand. Can you give more details?

Reply:
I don't think that there is any risk for circular references.